src/libs/kata-types/src/annotations/mod.rs

```diff
@@ -12,3 +23,799 @@ pub mod crio;
+                    Some(nums)
+                } else {
+                    None
+                }
+            }
+
+            Err(e) => {
+                warn!(
+                    sl!(),
+                    "failed to parse u32 value from {}, error: {:?}", s, e
+                );
+                None
+            }
+        }
+    }
+
+    /// Get the value of annotation with `key` as i32.
+    pub fn get_i32(&self, key: &str) -> Option<i32> {
+        let s = self.get(key)?;
+        match s.parse::<i32>() {
+            Ok(nums) => {
+                if nums > 0 {
+                    Some(nums)
+                } else {
+                    None
+                }
+            }
+
+            Err(e) => {
+                warn!(
+                    sl!(),
+                    "failed to parse u32 value from {}, error: {:?}", s, e
+                );
+                None
+            }
+        }
+    }
+
+    /// Get the value of annotation with `key` as u64.
+    pub fn get_u64(&self, key: &str) -> Option<u64> {
+        let s = self.get(key)?;
+        match s.parse::<u64>() {
+            Ok(nums) => {
+                if nums > 0 {
+                    Some(nums)
+                } else {
+                    None
+                }
+            }
+
+            Err(e) => {
+                warn!(
+                    sl!(),
+                    "failed to parse u64 value from {}, error: {:?}", s, e
+                );
+                None
+            }
+        }
+    }
+    /*
+    /// add annotation for config
+    pub fn add_annotation() -> Reuslt<()>{
+
+    }
+    */
+}
+
+// Miscellaneous annotations.
+impl Annotation {
+    /// Get the annotation of sandbox configuration file path.
+    pub fn get_sandbox_config_path(&self) -> Option<String> {
+        self.get(SANDBOX_CONFIG_PATH_KEY)
+    }
+
+    /// Get the annotation of bundle path.
+    pub fn get_bundle_path(&self) -> Option<String> {
+        self.get(BUNDLE_PATH_KEY)
+    }
+
+    /// Get the annotation of container type.
+    pub fn get_container_type(&self) -> Option<String> {
+        self.get(CONTAINER_TYPE_KEY)
+    }
+
+    /// Get the annotation to specify the Resources.Memory.Swappiness.
+    pub fn get_container_resource_swappiness(&self) -> Option<u32> {
+        let v = self.get_u32(KATA_ANNO_CONTAINER_RESOURCE_SWAPPINESS)?;
+        if v > 100 {
+            None
+        } else {
+            Some(v)
+        }
+    }
+
+    /// Get the annotation to specify the Resources.Memory.Swap.
+    pub fn get_container_resource_swap_in_bytes(&self) -> Option<String> {
+        self.get(KATA_ANNO_CONTAINER_RESOURCE_SWAP_IN_BYTES)
+    }
+}
+
+impl Annotation {
+    /// update config info by annotation
+    pub fn update_config_by_annotation(
+        &self,
+        config: &mut TomlConfig,
+        hypervisor_name: &str,
+        agent_name: &str,
+    ) -> Result<()> {
+        if let Some(hv) = config.hypervisor.get_mut(hypervisor_name) {
```

Comment:
Please use a [early return pattern](https://medium.com/swlh/return-early-pattern-3d18a41bba8) to reduce too many nested if/loop and avoid long function body.

Reply:
fixed in next version